amgut/lib/data_access/ag_data_access.py

```diff
@@ -380,40 +389,50 @@ def getAnimalParticipants(self, ag_login_id):
                     sql, [ag_login_id, 2])]
 
     def getParticipantSamples(self, ag_login_id, participant_name):
-        results = self._sql.execute_proc_return_cursor(
-            'ag_get_participant_samples', [ag_login_id, participant_name])
-        rows = results.fetchall()
-        col_names = self._get_col_names_from_cursor(results)
-        results.close()
+        sql = """SELECT  barcode, site_sampled, sample_date, sample_time,
+                    notes, status
+                 FROM ag_kit_barcodes akb
+                 INNER JOIN barcode USING (barcode)
+                 INNER JOIN ag_kit ak USING (ag_kit_id)
+                 WHERE (site_sampled IS NOT NULL AND site_sampled::text <> '')
+                 AND ag_login_id = %s AND participant_name = %s"""
```

Comment:
sql engines are works of voodoo

On Mon, Aug 24, 2015 at 3:16 PM, Joshua Shorenstein <
notifications@github.com> wrote:

> In amgut/lib/data_access/ag_data_access.py
> <https://github.com/biocore/american-gut-web/pull/465#discussion_r37805842>
> :
>
> > @@ -380,40 +389,50 @@ def getAnimalParticipants(self, ag_login_id):
> >                      sql, [ag_login_id, 2])]
> >
> >      def getParticipantSamples(self, ag_login_id, participant_name):
> > -        results = self._sql.execute_proc_return_cursor(
> > -            'ag_get_participant_samples', [ag_login_id, participant_name])
> > -        rows = results.fetchall()
> > -        col_names = self._get_col_names_from_cursor(results)
> > -        results.close()
> > +        sql = """SELECT  barcode, site_sampled, sample_date, sample_time,
> > +                    notes, status
> > +                 FROM ag_kit_barcodes akb
> > +                 INNER JOIN barcode USING (barcode)
> > +                 INNER JOIN ag_kit ak USING (ag_kit_id)
> > +                 WHERE (site_sampled IS NOT NULL AND site_sampled::text <> '')
> > +                 AND ag_login_id = %s AND participant_name = %s"""
>
> Where clause order doesn't matter in SQL, since the SQL parser will
> optimize the order itself as part of running the query. See here
> <http://stackoverflow.com/questions/11436469/does-order-of-where-clauses-matter-in-sql>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/american-gut-web/pull/465/files#r37805842>.
>